ksql-engine/src/main/java/io/confluent/ksql/ddl/commands/DropSourceCommand.java

```diff
@@ -89,45 +87,34 @@ public DdlCommandResult run(MetaStore metaStore, boolean isValidatePhase) {
 
   private void deleteTopicIfNeeded(StructuredDataSource dataSource, boolean isValidatePhase) {
     if (!isValidatePhase && deleteTopic) {
-
-      executeWithRetries(new Callable<Void>() {
-        @Override
-        public Void call() throws Exception {
+      try {
+        ExecutorWithRetries.execute((Supplier<Future<Void>>) () -> {
```

Comment:
This won't currently retry: `kafkaTopicClient.deleteTopics` throws a `KsqlException` if anything goes wrong. This is not a `RetriableException` so the code won't retry it :(

Reply:
Fixed in ExecutorWithRetries